lib/ansible/modules/monitoring/zabbix/zabbix_host_facts.py

```diff
@@ -0,0 +1,245 @@
+            'filter': {
+                'ip': host_ips
+            }
+        })
+        if len(hostinterfaces) < 1:
+            self._module.fail_json(msg="Host not found: %s" % host_ips)
+        host_list = []
+        for hostinterface in hostinterfaces:
+            host = self._zapi.host.get({
+                'output': 'extend',
+                'selectGroups': 'extend',
+                'hostids': hostinterface['hostid']
+            })
+            host[0]['hostinterfaces'] = hostinterface
+            host_list.append(host[0])
+        return host_list
+
+    def delete_duplicate_hosts(self, hosts):
+        """ Delete duplicated hosts """
+        unique_hosts = []
+        listed_hostnames = []
+        for zabbix_host in hosts:
+            if zabbix_host['name'] in listed_hostnames:
+                self._zapi.host.delete([zabbix_host['hostid']])
+                continue
+            unique_hosts.append(zabbix_host)
+            listed_hostnames.append(zabbix_host['name'])
+        return unique_hosts
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            server_url=dict(type='str', required=True, aliases=['url']),
+            login_user=dict(type='str', required=True),
+            login_password=dict(type='str', required=True, no_log=True),
+            host_name=dict(type='str', default='', required=False),
+            host_ip=dict(type='list', default=[], required=False),
+            http_login_user=dict(type='str', required=False, default=None),
+            http_login_password=dict(type='str', required=False, default=None, no_log=True),
+            timeout=dict(type='int', default=10),
+            exact_match=dict(type='bool', required=False, default=False),
+            remove_duplicate=dict(type='bool', required=False, default=False)
+        ),
+        supports_check_mode=True
+    )
+
+    if not HAS_ZABBIX_API:
+        module.fail_json(msg="Missing required zabbix-api module (check docs or install with: pip install zabbix-api)")
+
+    server_url = module.params['server_url']
+    login_user = module.params['login_user']
+    login_password = module.params['login_password']
+    http_login_user = module.params['http_login_user']
+    http_login_password = module.params['http_login_password']
+    host_name = module.params['host_name']
+    host_ips = module.params['host_ip']
+    timeout = module.params['timeout']
+    exact_match = module.params['exact_match']
+    is_remove_duplicate = module.params['remove_duplicate']
+
+    zbx = None
+    # login to zabbix
+    try:
+        zbx = ZabbixAPIExtends(server_url, timeout=timeout, user=http_login_user, passwd=http_login_password)
+        zbx.login(login_user, login_password)
+    except Exception as e:
+        module.fail_json(msg="Failed to connect to Zabbix server: %s" % e)
+
+    host = Host(module, zbx)
+
+    if host_name:
+        is_host_exist = host.is_host_exist(host_name, exact_match)
```

Comment:
This is unnecessary as error checking is handled in `get_hosts_by_host_name` as well. I think you can remove it completely with the whole `is_host_exist` method to save a few lines of code. Basically you want to do the same as with `get_hosts_by_ip`.